### PR TITLE
Handle more panics

### PR DESCRIPTION
**Pull request by MDrakos**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-3136" title="DX-3136" target="_blank">BugDX-3136</a>  Rollbar: Could not report messages as CheckMessages return an error: Error sending messages request:
  </td></table>
  <br />

### CODE REVIEW

File: internal/poller/poller.go [hunk 1]
```diff
@@ -19,8 +21,15 @@ type Poller struct {
 }
 
 func New(interval time.Duration, pollFunc func() (interface{}, error)) *Poller {
+	wrappedFn := func() (interface{}, error) {
+		defer func() {
+			panics.LogAndPanic(recover(), debug.Stack())
+		}()
+		return pollFunc()
+	}
```
Comment (MDrakos) [inline internal/poller/poller.go:29]: Panics can occur in the goroutine where the poll func is started. If we don't want this automatic wrapping we can update each caller of this library to manually inject it into their pollfunc.

File: cmd/state-svc/internal/messages/messages.go [hunk 2]
```diff
@@ -74,7 +74,11 @@ func (m *Messages) Check(command string, flags []string) ([]*graph.MessageInfo,
 	if cacheValue == nil {
 		return []*graph.MessageInfo{}, nil
 	}
-	allMessages := cacheValue.([]*graph.MessageInfo)
+
+	allMessages, ok := cacheValue.([]*graph.MessageInfo)
+	if !ok {
+		return nil, errs.New("Could not get messages from cache")
```
Comment (Naatan) [inline cmd/state-svc/internal/messages/messages.go:80]:
```suggestion
		return nil, errs.New("cacheValue has unexpected type: %T", cacheValue)
```

File: cmd/state-svc/internal/messages/messages.go [hunk 3]
```diff
@@ -110,7 +114,11 @@ func check(params *ConditionParams, messages []*graph.MessageInfo, lastReportMap
 		logging.Debug("Checking message %s", message.ID)
 		// Ensure we don't show the same message too often
 		if lastReport, ok := lastReportMap[message.ID]; ok {
-			lastReportTime, err := time.Parse(time.RFC3339, lastReport.(string))
+			lr, ok := lastReport.(string)
+			if !ok {
+				return nil, errs.New("Could not get last reported time for message %s as it's not a string", message.ID)
```
Comment (Naatan) [inline cmd/state-svc/internal/messages/messages.go:119]:
```suggestion
				return nil, errs.New("Could not get last reported time for message %s as it's not a string: %T", message.ID, lastReport)
```

File: internal/poller/poller.go (hunk 1 above)
Comment (Naatan) [inline internal/poller/poller.go:29]: I think we need to implement this as part of the pollFunc that we pass. Reason being that if we catch it here the stacktrace won't point to what caused the panic.

**Review by Naatan — CHANGES_REQUESTED** [review]

**Review by Naatan — APPROVED** [review]

**Pull request merged**